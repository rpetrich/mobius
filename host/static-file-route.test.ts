import { brotliedBufferFromRoute, gzippedBufferFromRoute, staticFileRoute, stringFromRoute } from "./static-file-route";

const html = `<html><head>Test</head><body>body</body></html>`;

describe("staticFileRoute", () => {
	it("string should match snapshot", () => {
		expect(staticFileRoute("/test.html", html)).toMatchSnapshot();
	});
	it("buffer should match snapshot", () => {
		expect(staticFileRoute("/test.html", Buffer.from(html))).toMatchSnapshot();
	});
});

describe("stringFromRoute", () => {
	it("string should match", () => {
		expect(stringFromRoute(staticFileRoute("/test.html", html))).toBe(html);
	});
	it("buffer should match", () => {
		expect(stringFromRoute(staticFileRoute("/test.html", Buffer.from(html)))).toBe(html);
	});
});

describe("gzippedBufferFromRoute", () => {
	it("buffer should match", () => {
		expect(gzippedBufferFromRoute(staticFileRoute("/test.html", html))).toMatchSnapshot();
	});
});

describe("brotliedBufferFromRoute", () => {
	it("buffer should match", () => {
		expect(brotliedBufferFromRoute(staticFileRoute("/test.html", html))).toMatchSnapshot();
	});
});
