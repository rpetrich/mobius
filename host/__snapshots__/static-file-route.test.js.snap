// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brotliedBufferFromRoute buffer should match 1`] = `
Object {
  "data": Array [
    27,
    46,
    0,
    248,
    197,
    109,
    108,
    31,
    241,
    172,
    198,
    9,
    18,
    28,
    181,
    50,
    21,
    17,
    6,
    169,
    44,
    128,
    137,
    77,
    62,
    156,
    91,
    194,
    234,
    97,
    233,
    131,
    80,
    60,
  ],
  "type": "Buffer",
}
`;

exports[`gzippedBufferFromRoute buffer should match 1`] = `
Object {
  "data": Array [
    31,
    139,
    8,
    0,
    0,
    0,
    0,
    0,
    2,
    19,
    179,
    201,
    40,
    201,
    205,
    177,
    179,
    201,
    72,
    77,
    76,
    177,
    11,
    73,
    45,
    46,
    177,
    209,
    7,
    51,
    109,
    146,
    242,
    83,
    42,
    237,
    64,
    132,
    141,
    62,
    152,
    9,
    20,
    7,
    41,
    4,
    0,
    57,
    101,
    200,
    204,
    47,
    0,
    0,
    0,
  ],
  "type": "Buffer",
}
`;

exports[`staticFileRoute buffer should match snapshot 1`] = `
Object {
  "buffer": Object {
    "data": Array [
      60,
      104,
      116,
      109,
      108,
      62,
      60,
      104,
      101,
      97,
      100,
      62,
      84,
      101,
      115,
      116,
      60,
      47,
      104,
      101,
      97,
      100,
      62,
      60,
      98,
      111,
      100,
      121,
      62,
      98,
      111,
      100,
      121,
      60,
      47,
      98,
      111,
      100,
      121,
      62,
      60,
      47,
      104,
      116,
      109,
      108,
      62,
    ],
    "type": "Buffer",
  },
  "etag": "\\"2f-ElLqJBdvCIMM8vS0gUFUrF59SGA\\"",
  "foreverPath": "/test.RdozNLuQSTeQJxfw.html",
  "integrity": "sha256-RdozNLuQSTeQJxfwgx9c+6AQihjghqCXYvlog/Cjews=",
  "path": "/test.html",
}
`;

exports[`staticFileRoute string should match snapshot 1`] = `
Object {
  "buffer": Object {
    "data": Array [
      60,
      104,
      116,
      109,
      108,
      62,
      60,
      104,
      101,
      97,
      100,
      62,
      84,
      101,
      115,
      116,
      60,
      47,
      104,
      101,
      97,
      100,
      62,
      60,
      98,
      111,
      100,
      121,
      62,
      98,
      111,
      100,
      121,
      60,
      47,
      98,
      111,
      100,
      121,
      62,
      60,
      47,
      104,
      116,
      109,
      108,
      62,
    ],
    "type": "Buffer",
  },
  "etag": "\\"2f-ElLqJBdvCIMM8vS0gUFUrF59SGA\\"",
  "foreverPath": "/test.RdozNLuQSTeQJxfw.html",
  "integrity": "sha256-RdozNLuQSTeQJxfwgx9c+6AQihjghqCXYvlog/Cjews=",
  "path": "/test.html",
  "string": "<html><head>Test</head><body>body</body></html>",
}
`;
